### PR TITLE
feat: Hide HyperJson buttons when selecting value

### PR DESCRIPTION
**Pull request by ernestii**
https://github.com/hyperdxio/hyperdx/assets/20255948/c6ff3bb8-6df2-4be2-8c58-49a91b23429e

### CODE REVIEW

File: packages/app/src/components/HyperJson.tsx [hunk 1]
```diff
@@ -177,6 +195,25 @@ const Line = React.memo(
...
+    const valueRef = React.useRef<HTMLSpanElement>(null);
+    const [isSelectingValue, setIsSelectingValue] = React.useState(false);
+    const handleValueSelectStart = React.useCallback(() => {
+      setIsSelectingValue(true);
+    }, []);
+    const handleValueMouseUp = React.useCallback(() => {
+      setIsSelectingValue(false);
+    }, []);
+    React.useEffect(() => {
+      const _valueRef = valueRef.current;
```
Comment (colehpage) [inline packages/app/src/components/HyperJson.tsx:208]: Should this be optionally chained since valueRef can be null? or is that case impossible

**Review by colehpage — APPROVED** [review]
I don't think this every bothered me 😆 did it get in the way on smaller screens or something? regardless 👍

File: packages/app/src/components/HyperJson.tsx (hunk 1 above)
Comment (ernestii) [reply packages/app/src/components/HyperJson.tsx:208]: i believe it can't be null since it comes from useRef, but only .current can be null?

Comment (colehpage) [reply packages/app/src/components/HyperJson.tsx:208]: No I think you're right, I still mess up useref hooks lifecycle 😓

**Pull request merged**